sql-plugin/src/main/scala/com/nvidia/spark/rapids/ParquetCachedBatchSerializer.scala

```diff
@@ -860,11 +867,17 @@ protected class ParquetCachedBatchSerializer extends GpuCachedBatchSerializer {
         if (!cbIter.hasNext) {
           Iterator.empty
         } else {
-          new ShimCurrentBatchIterator(
-            cbIter.next().asInstanceOf[ParquetCachedBatch],
-            conf,
-            selectedAttributes,
-            options, hadoopConf)
+          try {
+            new ShimCurrentBatchIterator(
+              cbIter.next().asInstanceOf[ParquetCachedBatch],
+              conf,
+              selectedAttributes,
+              options, hadoopConf)
+          } catch {
+            // The CurrentBatchIterator encountered an empty batch. Return an empty Iterator just
+            // like we do when hasNext returns false
+            case _: EmptyBatchEncounteredException => Iterator.empty
```

Comment:
I don't think this is correct. An empty batch does not mean that there is nothing to read. It means that there are no columns, but we still need to return rows.